lib/pos_blockchain.go

```diff
@@ -831,11 +835,41 @@ func (bc *Blockchain) isBlockTimestampValidRelativeToParentPoS(header *MsgDeSoHe
 	return nil
 }
 
+// isBlockTimestampTooFarInFuturePoS validates that the block's timestamp is not too far in the future based
+// on the configured block timestamp drift.
+//
+// We use the snapshotted global params to validate that the block's timestamp isn't too far ahead in the
+// future. We use the snapshotted global params specifically so that the drift timestamp check behaves
+// consistently even for orphan blocks that are 1 epoch in the future..
+func (bc *Blockchain) isBlockTimestampTooFarInFuturePoS(header *MsgDeSoHeader) (bool, error) {
+	utxoView, err := NewUtxoView(bc.db, bc.params, nil, bc.snapshot, nil)
+	if err != nil {
+		return false, errors.Wrap(err, "isBlockTimestampTooFarInFuturePoS: Problem initializing UtxoView")
+	}
+
+	simulatedEpochEntryForBlock, err := utxoView.SimulateAdjacentEpochEntryForBlockHeight(header.Height)
```

Comment:
Ah okay this won't work if we're blocksyncing and we've got a block that's more than 1 epoch in the past. I think a very simple solution to this would be to simply check if the header's timestamp is less than the current timestamp at the top of the function. This way we don't even need to create a view or compute the snapshot at epoch number. Basically, if the block is in the past, we can just exit early and say that the block is certainly not too far in the future.

Reply:
Great suggestion. Done